test/e2e/framework/pods.go

```diff
@@ -70,3 +74,23 @@ func (f *Framework) MungePodSpec(pod *api.Pod) {
 		pod.Spec.NodeName = TestContext.NodeName
 	}
 }
+
+func (f *Framework) UpdatePod(name string, update func(pod *api.Pod)) {
+	ExpectNoError(wait.Poll(time.Millisecond*500, time.Second*30, func() (bool, error) {
+		pod, err := f.PodClient().Get(name)
+		if err != nil {
+			return false, fmt.Errorf("failed to get pod: %v", err)
+		}
+		update(pod)
+		_, err = f.PodClient().Update(pod)
+		if err == nil {
+			Logf("Successfully updated pod")
+			return true, nil
+		}
+		if errors.IsConflict(err) {
+			Logf("Conflicting update to pod, re-get and re-update: %v", err)
+			return false, nil
+		}
+		return false, fmt.Errorf("failed to update pod: %v", err)
```

Comment:
nit: not sure if the `err` contains the pod information. Print the pod name just in case.